lib/charms/layer/jenkins/configuration.py

```diff
@@ -1,4 +1,5 @@
 import os
+from urllib.parse import urlparse
```

Comment:
That does conform to pep8, urllib is part of the standard library and it is grouped with os also part of the standard library with relative imports coming second.